### PR TITLE
Improve buildfarm tools lib

### PR DESCRIPTION
**Pull request by Crola1702**
# Description

This PR creates a library to interact with SQL scripts from `database/scripts` folder. It will simplify the process of filtering buildfarmer.db data

* Rewrite `buildfarm_tools.rb` script work exposing methods that interact with buildfarm database resource throughout `sql_run.sh` script.
* Adapt `check_buildfarm.rb` to work with new library script.
* Add `-header` option in `sql_run.sh` to show headers of all queries

### CODE REVIEW

File: database/scripts/test_regression_history.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+SELECT test_failures.error_name,
+    test_failures.job_name,
```
Comment (Blast545) [inline database/scripts/test_regression_history.sql:2]: The sql is alright, but I think the name of the should reflect your usage of the param1 and param2, with "test_regression_history" it's not straightforward to me what this sql should do.

File: database/scripts/lib/buildfarm_tools.rb [hunk 2]
```diff
@@ -1,40 +1,116 @@
...
+
+module BuildfarmToolsLib
+  class BuildfarmToolsError < RuntimeError; end
+
+  KNOWN_REASONS = Set.new([
+                            'Jenkins channel closing down (Agent disconnected)',
+                            'Network error: Failed to clone github repo'
+                          ])
+
+  CONSISTENT_THRESHOLD = 3
```
Comment (Blast545) [inline database/scripts/lib/buildfarm_tools.rb:13]: Maybe `consecutive ` instead of consistent?

File: database/scripts/lib/buildfarm_tools.rb [hunk 3]
```diff
@@ -1,40 +1,116 @@
...
+  FLAKY_BUILDS_THRESHOLD = 3 # For 15 days
```
Comment (Blast545) [inline database/scripts/lib/buildfarm_tools.rb:14]: Maybe you also want to define the 15 days as a constant? Something along the lines of `FLAKY_BUILDS_DAYS_RANGE`

File: database/scripts/lib/buildfarm_tools.rb (hunk 2 above)
Comment (Blast545) [reply database/scripts/lib/buildfarm_tools.rb:13]: Same below when "consistent" is used.

**Review by Blast545 — COMMENTED** [review]
Some nits, otherwise lgtm

File: database/scripts/test_regression_history.sql (hunk 1 above)
Comment (Crola1702) [reply database/scripts/test_regression_history.sql:2]: WDYT about: `error_appearances_in_job`?

File: database/scripts/lib/buildfarm_tools.rb (hunk 3 above)
Comment (Crola1702) [reply database/scripts/lib/buildfarm_tools.rb:14]: This is parametrized in all the code, so there is no hardcoded value for the flaky builds days range. I can add it as a default value instead

File: database/scripts/test_regression_history.sql (hunk 1 above)
Comment (Blast545) [reply database/scripts/test_regression_history.sql:2]: SGTM!

**Review by Blast545 — APPROVED** [review]
LGTM, pls address https://github.com/osrf/buildfarm-tools/pull/60#discussion_r1657732822 before merging

**Review by Blast545 — APPROVED** [review]

**Pull request merged**